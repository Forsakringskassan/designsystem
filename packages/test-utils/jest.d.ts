export * from "./dist/types/jest";

declare global {
    namespace jest {
        interface Matchers<R> {
            /**
             * Expects element to have focus.
             */
            toHaveFocus(): R;
        }
    }
}
