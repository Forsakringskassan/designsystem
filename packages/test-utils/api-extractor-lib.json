{
    "$schema": "https://developer.microsoft.com/json-schemas/api-extractor/v7/api-extractor.schema.json",
    "extends": "../../api-extractor.json",
    "mainEntryPointFilePath": "<projectFolder>/temp/types/lib.d.ts",
    "apiReport": {
        "enabled": false
    },
    "dtsRollup": {
        "publicTrimmedFilePath": "<projectFolder>/dist/types/lib.d.ts"
    }
}
