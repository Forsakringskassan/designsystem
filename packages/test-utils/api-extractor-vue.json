{
    "$schema": "https://developer.microsoft.com/json-schemas/api-extractor/v7/api-extractor.schema.json",
    "extends": "../../api-extractor.json",
    "mainEntryPointFilePath": "<projectFolder>/temp/types/vue/index.d.ts",
    "apiReport": {
        "enabled": false
    },
    "dtsRollup": {
        "publicTrimmedFilePath": "<projectFolder>/dist/types/vue.d.ts"
    }
}
