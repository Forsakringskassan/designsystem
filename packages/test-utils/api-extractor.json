{
    "$schema": "https://developer.microsoft.com/json-schemas/api-extractor/v7/api-extractor.schema.json",
    "extends": "../../api-extractor.json",
    "tsdocMetadata": {
        "tsdocMetadataFilePath": "<projectFolder>/dist/types/tsdoc-metadata.json"
    },
    "dtsRollup": {
        "enabled": false
    }
}
