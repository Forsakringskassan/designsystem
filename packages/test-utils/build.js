const fs = require("node:fs/promises");
const path = require("node:path");
const esbuild = require("esbuild");

const type = {
    cjs: "commonjs",
    esm: "module",
};

async function build() {
    for (const format of ["cjs", "esm"]) {
        const outdir = `dist/${format}`;
        const common = {
            logLevel: "info",
            sourcemap: true,
            external: ["vue", "@fkui/*"],
            bundle: true,
            platform: "node",
            format,
            target: "node20",
            outdir,
        };
        await esbuild.build({
            ...common,
            entryPoints: ["src/lib.ts", "src/jest.ts"],
        });
        await esbuild.build({
            ...common,
            entryPoints: [{ in: "src/vue/index.ts", out: "vue" }],
            define: {
                "process.env.NODE_ENV": JSON.stringify("development"),
            },
        });
        const pkg = JSON.stringify({ type: type[format] }, null, 2);
        await fs.writeFile(path.join(outdir, "package.json"), pkg, "utf-8");
    }
}

build().catch((err) => {
    console.error(err);
});
