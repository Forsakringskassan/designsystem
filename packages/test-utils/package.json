{
  "name": "@fkui/test-utils",
  "version": "6.6.0",
  "description": "FKUI test utils for Jest",
  "keywords": [
    "fkui",
    "designsystem",
    "test",
    "jest",
    "vue",
    "cypress"
  ],
  "homepage": "https://designsystem.forsakringskassan.se/",
  "bugs": "https://github.com/Forsakringskassan/designsystem/issues",
  "repository": {
    "type": "git",
    "url": "git+https://github.com/Forsakringskassan/designsystem.git",
    "directory": "packages/test-utils"
  },
  "license": "MIT",
  "author": "Försäkringskassan",
  "sideEffects": [
    "./dist/cjs/jest.js",
    "./dist/esm/jest.js",
    "./src/matchers/augmentations.ts",
    "./src/matchers/index.ts"
  ],
  "exports": {
    ".": {
      "types": "./dist/types/lib.d.ts",
      "import": "./dist/esm/lib.js",
      "require": "./dist/cjs/lib.js"
    },
    "./jest": {
      "types": "./dist/types/jest.d.ts",
      "import": "./dist/esm/jest.js",
      "require": "./dist/cjs/jest.js"
    },
    "./vue": {
      "types": "./dist/types/vue.d.ts",
      "import": "./dist/esm/vue.js",
      "require": "./dist/cjs/vue.js"
    }
  },
  "main": "dist/cjs/lib.js",
  "types": "dist/types/lib.d.ts",
  "files": [
    "dist",
    "jest.d.ts",
    "vue.d.ts"
  ],
  "scripts": {
    "prebuild": "rimraf dist temp && tsc",
    "build": "node build.js",
    "postbuild": "fk-api-extractor --patch-augmentations api-extractor.json api-extractor-*.json",
    "clean": "rimraf .jest-cache dist temp test-results",
    "prepack": "release-prepack --bundle --retain-scripts",
    "postpack": "release-postpack",
    "prepublishOnly": "release-prepublish --bundle --retain-scripts",
    "test": "jest"
  },
  "jest": {
    "preset": "@forsakringskassan/jest-config-vue"
  },
  "devDependencies": {
    "@vue/server-renderer": "3.5.13",
    "@vue/test-utils": "2.4.6",
    "esbuild": "0.25.3",
    "jest-environment-jsdom": "29.7.0",
    "vue": "3.5.13"
  },
  "peerDependencies": {
    "jest": "^26.0.0 || ^27.0.0 || ^28.0.0 || ^29.0.0",
    "vue": "^3.0.0"
  },
  "peerDependenciesMeta": {
    "vue": {
      "optional": true
    }
  },
  "engines": {
    "node": ">= 20",
    "npm": ">= 7"
  }
}
