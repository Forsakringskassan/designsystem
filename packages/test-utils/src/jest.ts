/* imported for side-effects only */
import "./matchers";

export * from "./lib";
