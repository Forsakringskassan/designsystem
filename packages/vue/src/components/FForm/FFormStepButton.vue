<template>
    <button
        :id="id"
        class="button button--margin-bottom-0 button--large"
        type="button"
        v-bind="$attrs"
        :aria-expanded="isOpen ? 'true' : 'false'"
        data-form-step-button
    >
        <template v-if="isOpen">
            <f-icon name="success" class="button__icon" />
            {{ $t("fkui.form-step-button.close", "OK") }}
        </template>
        <template v-else>
            <f-icon name="pen" class="button__icon" />
            {{
                isAnyFieldTouched
                    ? $t("fkui.form-step-button.edit", "Ã„ndra")
                    : $t("fkui.form-step-button.open", "Fyll i")
            }}
        </template>
        <span class="sr-only">{{ additionalScreenreaderText }}</span>
    </button>
</template>

<script lang="ts">
import { defineComponent } from "vue";
import { ElementIdService } from "@fkui/logic";
import { FIcon } from "../FIcon";
import { TranslationMixin } from "../../plugins";

/**
 * @deprecated `<f-form-step-button>` is deprecated.
 * See migration guide: https://forsakringskassan.github.io/latest/components/fform.html
 */
export default defineComponent({
    name: "FFormStepButton",
    components: { FIcon },
    mixins: [TranslationMixin],
    inheritAttrs: false,
    model: {
        prop: "sr-only",
    },
    props: {
        /**
         * If the id is not set a random value will be generated.
         */
        id: {
            type: String,
            required: false,
            default: () => ElementIdService.generateElementId(),
        },
        isOpen: Boolean,
        isAnyFieldTouched: Boolean,
        additionalScreenreaderText: {
            type: String,
            required: false,
            default: "",
        },
    },
});
</script>
