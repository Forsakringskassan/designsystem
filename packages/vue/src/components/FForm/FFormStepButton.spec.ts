import { VueWrapper, mount } from "@vue/test-utils";
import flushPromises from "flush-promises";
import FFormStepButton from "./FFormStepButton.vue";
import "html-validate/jest";

function createWrapper({
    props = {},
    slots = {},
    attrs = {},
    options = {},
} = {}): VueWrapper {
    return mount(FFormStepButton, {
        attrs: { ...attrs },
        props: { ...props },
        slots: { ...slots },
        ...options,
        global: {
            stubs: [],
        },
    });
}

describe("snapshots", () => {
    it("should match snapshot when pristine", async () => {
        const wrapper = createWrapper();
        await flushPromises();

        expect(wrapper.element).toMatchSnapshot();
    });

    it.each`
        isOpen   | isAnyFieldTouched | buttonText  | iconName
        ${false} | ${false}          | ${"Fyll i"} | ${"pen"}
        ${false} | ${true}           | ${"Ã„ndra"}  | ${"pen"}
        ${true}  | ${true}           | ${"OK"}     | ${"success"}
        ${true}  | ${false}          | ${"OK"}     | ${"success"}
    `(
        'should have the text "$buttonText" and icon name "$iconName" if the formStep is open "$isOpen" and is edited "$isAnyFieldTouched"',
        async ({ isOpen, isAnyFieldTouched, buttonText, iconName }) => {
            const wrapper = createWrapper({
                props: {
                    isOpen,
                    isAnyFieldTouched,
                },
            });
            await flushPromises();

            //            wrapper.vm.$props.isOpen = isOpen;
            //            wrapper.vm.$props.isAnyFieldTouched = isAnyFieldTouched;
            await wrapper.vm.$nextTick();

            expect(wrapper.get("button").text()).toBe(buttonText);
            expect(wrapper.get("svg").classes()).toContain("button__icon");
            expect(wrapper.get("svg").classes()).toContain(
                `f-icon-${iconName}`,
            );
        },
    );
});
