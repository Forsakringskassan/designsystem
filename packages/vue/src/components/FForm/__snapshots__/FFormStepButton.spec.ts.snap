// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshots should match snapshot when pristine 1`] = `
<button
  aria-expanded="false"
  class="button button--margin-bottom-0 button--large"
  data-form-step-button=""
  id="fkui-vue-element-0001"
  type="button"
>
  
  <svg
    aria-hidden="true"
    class="button__icon icon f-icon-pen"
    focusable="false"
  >
    
    
    <use
      xlink:href="#f-icon-pen"
    />
  </svg>
   Fyll i
  
  <span
    class="sr-only"
  />
</button>
`;
